tests/tests_e3/os/process/main_test.py

```diff
@@ -115,6 +116,123 @@ def run_test():
         e.restore()
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="A linux test")
+def test_rlimit_ctrl_c():
+    """Test rlimit CTRL-C.
+
+    When a parent process launched two or more child processes using rlimit, the CTRL-C
+    command no longer worked.
+
+    This was because when an rlimit process was launched, it became the foreground
+    process.
+
+    However, when the foreground process was killed, it left the parent process without
+    a foreground, so CTRL-C was ignored.
+
+    Examples:
+       Example 1: Spawn 1 rlimit child process:
+            python (Parent process)
+            |
+            -> rlimit (Child process / foreground process)
+
+            A CTRL-C appear:
+                The child process in the foreground was killed, leaving the parent
+                process alone with no foreground process but no child. Due to our
+                usage, this posed no known problems. However, leaving the parent
+                process without a foreground can cause unexpected results.
+
+       Example 2: Spawn 2 rlimit child process:
+            python (Parent process)
+            |
+            -> rlimit (Child process) ==> This process was the foreground process as
+            |                             long as no other rlimit process was running.
+            |                             In this example, we have 2 rlimit processes,
+            |                             so this process is not in the foreground.
+            |
+            -> rlimit (Child process / foreground process)
+
+            A CTRL-C appear:
+                The foreground process has been killed, leaving no foreground process.
+                Signals were no longer propagated, so CTRL-C did nothing.
+    """
+    try:
+        from ptyprocess import PtyProcess
+    except ImportError:
+        raise ImportError("ptyprocess is needed to run this tests") from None
+
+    # Only a linux test
+    assert sys.platform.startswith("linux"), "This test make sens only on linux"
+
+    script_to_run = """
+from __future__ import annotations
```

Comment:
I do not think this is needed since there are no annotation in this python scriptlet.

Reply:
Not really necessary but it's good practice so I use it every time.